File: app/src/main/java/com/example/delhitransit/Activities/BusListAdapter.java
package com.example.delhitransit.Activities;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.delhitransit.RoomData.DataClasses.BusPositionUpdate;
import com.example.delhitransit.R;

import java.util.List;

public class BusListAdapter extends RecyclerView.Adapter<BusListAdapter.MyViewHolder> {

    private List<BusPositionUpdate> mDataset;

    public class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView textView;

        public MyViewHolder(TextView v) {
            super(v);
            textView = v;
        }
    }

    // Create new views (invoked by the layout manager)
    @Override
    public BusListAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        // create a new view
        TextView v = (TextView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_bus, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        holder.textView.setText(mDataset.get(position).toString());
    }

    @Override
    public int getItemCount() {
        if (mDataset != null) {
            return mDataset.size();
        }
        return 0;
    }

    public void setDataset(List<BusPositionUpdate> mDataset) {
        this.mDataset = mDataset;
    }

    public BusListAdapter(List<BusPositionUpdate> myDataset) {
        mDataset = myDataset;
    }

}

File: app/src/main/java/com/example/delhitransit/Activities/BusListAdapter.kt
package com.example.delhitransit.Activities

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView

import androidx.recyclerview.widget.RecyclerView

import com.example.delhitransit.BusPositionUpdate
import com.example.delhitransit.R

class BusListAdapter(private var mDataset: List<BusPositionUpdate>?) : RecyclerView.Adapter<BusListAdapter.MyViewHolder>() {

    inner class MyViewHolder(// each data item is just a string in this case
            var textView: TextView) : RecyclerView.ViewHolder(textView)

    // Create new views (invoked by the layout manager)
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        // create a new view
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_bus, parent, false) as TextView

        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.textView.text = mDataset!![position].toString()
    }

    override fun getItemCount(): Int {
        return if (mDataset != null) {
            mDataset!!.size
        } else 0
    }

    fun setDataset(mDataset: List<BusPositionUpdate>) {
        this.mDataset = mDataset
    }

}